Tests/Shared.Specs/GenericCollectionAssertionOfStringSpecs.cs

```diff
@@ -2223,5 +2223,128 @@ public void When_using_StringCollectionAssertions_the_AndConstraint_should_have_
+            IEnumerable<string> collection = new string[] { "build succeded", "test failed" };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () => collection.Should().ContainMatch("* Failed");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            action.Should().Throw<XunitException>()
+                .WithMessage("Expected collection {\"build succeded\", \"test failed\"} to contain a match of \"* Failed\".");
+        }
+
+        [Fact]
+        public void When_asserting_empty_collection_for_match_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            IEnumerable<string> collection = new string[] { };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () => collection.Should().ContainMatch("* failed");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            action.Should().Throw<XunitException>()
+                .WithMessage("Expected collection {empty} to contain a match of \"* failed\".");
+        }
+
+        [Fact]
+        public void When_asserting_null_collection_for_match_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            IEnumerable<string> collection = null;
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () => collection.Should().ContainMatch("* failed");
```

Comment:
Please add `because` and `becauseArgs` to this test as well, as it hits another `FailWith` in the assertion implementation.